benchmarks/audition/fsdd.py

```diff
@@ -285,32 +333,18 @@ def run_resnet18():
     classes_space = list(combinations_45(nums, n_classes))
 
     # scale the data
-    x_spec = scale(x_spec.reshape(3000, -1), axis=1).reshape(3000, 32, 32)
+    x_spec = scale(x_spec.reshape(5400, -1), axis=1).reshape(5400, 32, 32)
     y_number = np.array(y_number)
 
     # need to take train/valid/test equally from each class
-    trainx = np.zeros((1, 32, 32))
-    trainy = np.zeros((1))
-    testx = np.zeros((1, 32, 32))
-    testy = np.zeros((1))
-    for i in range(10):
-        shuffler = np.random.permutation(300)
-        trainx = np.concatenate(
-            (trainx, x_spec[i * 300 : (i + 1) * 3000][shuffler][:240])
-        )
-        trainy = np.concatenate(
-            (trainy, y_number[i * 300 : (i + 1) * 3000][shuffler][:240])
-        )
-        testx = np.concatenate(
-            (testx, x_spec[i * 300 : (i + 1) * 3000][shuffler][240:])
-        )
-        testy = np.concatenate(
-            (testy, y_number[i * 300 : (i + 1) * 3000][shuffler][240:])
-        )
-    trainx = trainx[1:]
-    trainy = trainy[1:]
-    testx = testx[1:]
-    testy = testy[1:]
+    trainx, testx, trainy, testy = train_test_split(
+        x_spec,
+        y_number,
+        shuffle=True,
+        test_size=0.25,
+        train_size=0.50,
+        stratify=y_number,
+    )
```

Comment:
I don't feel that the size parameters are parsed correctly. Check shapes of indices and they might be an 3:1 split instead of 2:1.

Reply:
The shapes are as: trainx= (2700, 32, 32); testx =  (1350, 32, 32)  trainy = (2700,) testy = (1350,)

Reply:
but how could you access the validation sets?

Reply:
Sorry! I have corrected the code.